### PR TITLE
Docs fix/add introductory paragraphs to subsections

### LINKED ISSUE #4380: [Docs] Include introductory pragraphs
For specific subsections, TG has some introductory paragraphs, i.e. at https://core.telegram.org/bots/api#games. These are currently not available directly in the PTB docs.

We should

1. add them to the PTB docs (e.g. above paragrph to https://docs.python-telegram-bot.org/en/v21.4/telegram.payments-tree.html)
2. Revisit all "More info on xy »" links and make them point to the PTB docs where possible .That way, users don't have to jump between PTB and TG docs

### PR DESCRIPTION
**Pull request by mohdyusuf2312**
### Description

This pull request adds introductory paragraphs to several sections in the PTB documentation and updates the "More info on xy" links to point to the PTB docs where appropriate.

### Changes Made

- Added introductory paragraphs to the Payments, Passport, Games, Inline Mode, Stickers, and Available Types sections.
- Updated "More info on xy" links to point to relevant sections in the PTB docs.
- Added my name to the `AUTHORS.rst` file.

### Issue Addressed

docs/fix #4380 

### Available Types section before
image
### Available Types section after
image

### Stickers section before
image
### Stickers section after
image

### Inline Mode section before
image
### Inline Mode section after
image

### Payment section before
image
### Payment section after
image

### Games section before
image
### Games section after
image

### Passport section before
image
### Passport section after
image

### Additional Information

Please review the changes and let me know if any further adjustments are needed.

### CODE REVIEW

**Comment by Poolitzer** [comment]
Hello, thanks for the PR.

Please dont modify the gitignore. The available types change is not needed, neither do we provide json objects nor does python have 32 bit signed integers.

For the rest without looking deeply into it: Please use for linking to ours the internal links as we do in the docs. So e.g. paramref:`~telegram.Bot.local_mode` for a parameter, :attr:`telegram.Message.reply_markup` for an attribute, :meth:`telegram.Bot.set_game_score` for a method, etc.

**Comment by mohdyusuf2312** [comment]
image

I'm trying to apply internal linking, but it is not clickable. Can you please suggest how I can resolve this?

**Comment by Bibo-Joshi** [comment]
Hi. the linking is case sensitive, i.e. <code>:class:\`telegram.Game\`</code> can work, but <code>:class:\`Telegram.Game\`</code> can not. If you want to link to a headline, it should be possible to use the headline as lower case and "-" instead of " " as target - e.g. <code>:ref:\`inline-mode\`</code> for a headline "Inline Mode". I'm not 100% sure on this one though.
Additionally, when you build the documentation in the command line, you will see warnings and errors being printed for references that could not be resolved.
If you're new to documentation python code with Sphinx, please also have a look at their [documentation](https://www.sphinx-doc.org/en/master/), especiall at the [rst primer](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) and the [referencing syntax](https://www.sphinx-doc.org/en/master/usage/referencing.html).

**Comment by mohdyusuf2312** [comment]
> If you're new to documentation python code with Sphinx, please also have a look at their [documentation](https://www.sphinx-doc.org/en/master/), especiall at the [rst primer](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) and the [referencing syntax](https://www.sphinx-doc.org/en/master/usage/referencing.html).
Thanks @Bibo-Joshi Sir

File: docs/source/telegram.games-tree.rst [hunk 1]
```diff
@@ -1,6 +1,19 @@
 Games
 -----
 
+Your bot can offer users **HTML5 games** to play solo or to compete against each other in groups and one-on-one chats. Create games via `@BotFather <https://telegram.me/BotFather>`_ using the /*newgame* command. Please note that this kind of power requires responsibility: you will need to accept the terms for each game that your bots will be offering.
```
Comment (Bibo-Joshi) [inline docs/source/telegram.games-tree.rst:4]: `/*newgame*` will not render correctly. Please use <code>\`\`/newgame\`\`</code> to ensure that it's rendered as mono-spaced text.

File: docs/source/telegram.games-tree.rst [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+
+* Games are a new type of content on Telegram, represented by the :doc:`Game <telegram.game>` and :doc:`InlineQueryResultGame <telegram.inlinequeryresultgame>` objects.
```
Comment (Bibo-Joshi) [inline docs/source/telegram.games-tree.rst:6]: <code>:doc:\`Game <telegram.game>\`</code>

This is better, but is has two downsides:

* by using `:doc:`, Sphinx doesn't know that you're referencing a class. Indexing and rendering might be wrong. Please use `:class:` for classes `:meth:` for methods of classes etc. The available types are listed at https://www.sphinx-doc.org/en/master/usage/domains/python.html#python-xref-roles (drop the leading `:py`) and are also used throughout the PTB documentation
* you make your live harder by repeating "game" twice. If you write just <code>:class:\`telegram.Game\`</code> you aovid duplication and have correcting rendering.

File: docs/source/telegram.inline-tree.rst [hunk 3]
```diff
@@ -1,6 +1,12 @@
 Inline Mode
 -----------
 
+The following methods and objects allow your bot to work in `inline mode`_.
```
Comment (Bibo-Joshi) [inline docs/source/telegram.inline-tree.rst:4]: where does this link point to? Please make sure it points to https://core.telegram.org/bots/inline as it does in https://core.telegram.org/bots/api#inline-mode.

File: docs/source/telegram.inline-tree.rst [hunk 4]
```diff
@@ -1,6 +1,12 @@
...
+Please see our `Introduction to Inline bots <https://core.telegram.org/bots/inline>`_ for more details.
```
Comment (Bibo-Joshi) [inline docs/source/telegram.inline-tree.rst:5]:
```suggestion
Please see Telegrams `Introduction to Inline bots <https://core.telegram.org/bots/inline>`_ for more details.
```
kindly double check your changes for this kind of reference. We should make sure that with "we" we always mean PTB.

File: docs/source/telegram.inline-tree.rst [hunk 5]
```diff
@@ -1,6 +1,12 @@
...
+
+To enable this option, send the /setinline command to `@BotFather <https://t.me/botfather>`_ and provide the placeholder text that the user will see in the input field after typing your bot's name.
```
Comment (Bibo-Joshi) [inline docs/source/telegram.inline-tree.rst:7]:
```suggestion
To enable this option, send the ``/setinline`` command to `@BotFather <https://t.me/botfather>`_ and provide the placeholder text that the user will see in the input field after typing your bot's name.
```

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Thanks for the updates. I left a few comments. In addition, what's still missing is making the "more about" links ponit to the PTB docs rather than TG docs as explained in #4380, e.g. https://github.com/python-telegram-bot/python-telegram-bot/blob/422993b8ab05d49dd49ab58c500f880918d894a7/telegram/_reply.py#L95

**Comment by mohdyusuf2312** [comment]
Hello, is everything done, or should I do anything else?

File: telegram/_message.py [hunk 6]
```diff
@@ -359,7 +359,7 @@ class Message(MaybeInaccessibleMessage):
             about the animation. For backward compatibility, when this field is set, the document
             field will also be set.
         game (:class:`telegram.Game`, optional): Message is a game, information about the game.
-            `More about games >> <https://core.telegram.org/bots/api#games>`_.
+            `More about games >> <https://docs.python-telegram-bot.org/telegram.games-tree.html>`_.
```
Comment (Bibo-Joshi) [inline telegram/_message.py:362]: These should also work with sphinxs reference system. I guess <code>:ref:\`More about games >> \<games\>\`</code> should work, or `:doc:`. If you can not get the link to point to the correct location, you might need to introduce a refrence target manually - I'm sure that your favorite search engine will help you with that :)

File: docs/source/telegram.games-tree.rst [hunk 7]
```diff
@@ -1,14 +1,14 @@
...
+* You can manually add multiple buttons to your game message. Please note that the first button in the first row **must always** launch the game, using the field ``callback_game`` in :class:`telegram.InlineKeyboardButton`. You can add extra buttons according to taste: e.g., for a description of the rules, or to open the game's official community.
 * To make your game more attractive, you can upload a GIF animation that demostrates the game to the users via `BotFather <https://t.me/botfather>`_ (see `Lumberjack <https://t.me/gamebot?game=lumberjack>`_ for example).
-* A game message will also display high scores for the current chat. Use setGameScore to post high scores to the chat with the game, add the edit_message parameter to automatically update the message with the current scoreboard.
+* A game message will also display high scores for the current chat. Use setGameScore to post high scores to the chat with the game, add the ``edit_message`` parameter to automatically update the message with the current scoreboard.
```
Comment (Bibo-Joshi) [inline docs/source/telegram.games-tree.rst:11]:
```suggestion
* A game message will also display high scores for the current chat. Use :meth:`~telegram.Bot.set_game_score` to post high scores to the chat with the game, add the :paramref:`~telegram.Bot.set_game_score.paramref.edit_message` parameter to automatically update the message with the current scoreboard.
```

and similar in other places if you see any :)

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]

**Review by Bibo-Joshi — APPROVED** [review]
Added two small fixes that are not you're fault (see https://github.com/tdlib/telegram-bot-api/issues/617).
LGTM now. Thanks very much for the PR and your patience during the review process :)

**Comment by mohdyusuf2312** [comment]
Thank you sir, I feel very happy to complete my first contribution to open source. :)

**Pull request merged**